### DISCUSSION TITLE
Constrained Expansion with Boundary Conditions not working for individual meshes

### DISCUSSION THREAD

**Discussion by AryGP** [idaholab/moose#28154] [Q&A Modules: Solid mechanics]
Hey!
I am working on thermal expansion of a 2 cube system. I have 2 cubes stitched together, each set with their own subdomain ids. When I try to apply constraints on the individual cube I am not getting any expansion at all in Paraview as compared to applying the boundary conditions not specified by cube. I tried making the mesh cubes with a CombinerGenerator instead of a StitchedMeshGenerator and nothing changed. 

```
# 2 cube mesh (stitched)
[Mesh]
  [cube1]
    type = GeneratedMeshGenerator
    dim = 3
    nx = 2
    ny = 2
    nz = 2
    subdomain_name = 'cube1'
  []
  [cube2]
    type = GeneratedMeshGenerator
    dim = 3
    xmin = 1.2
    xmax = 2.2
    nx = 2
    ny = 2
    nz = 2
    subdomain_ids = '2'
    subdomain_name = 'cube2'
  []
  [stitch]
    type = StitchedMeshGenerator
    prevent_boundary_ids_overlap = False
    inputs = 'cube1 cube2'
    stitch_boundaries_pairs = 'right left'
  []
  #[combine]
   #type = CombinerGenerator
    #inputs = 'cube1 cube2'
  #[]
[]

[GlobalParams]
  displacements = 'disp_x disp_y disp_z'
  #code for constrained ex
  large_kinematics = false 
  eigenstrain_names = "thermal_contribution" 
[]

[AuxVariables]
  [./temp]
    initial_condition = 300.0
  [../]
  [./eigenstrain_yy]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./eigenstrain_xx]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./eigenstrain_zz]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./total_strain_yy]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./total_strain_xx]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./total_strain_zz]
    order = CONSTANT
    family = MONOMIAL
  [../]
[]

[Functions]
  [./temperature_load]
    type = ParsedFunction
    expression = t*(5000.0)+300.0
  [../]
[]

[Physics]
  [SolidMechanics]
    [QuasiStatic]
      [./all]
        strain = SMALL
        incremental = true
        add_variables = true
        eigenstrain_names = eigenstrain
        new_system = true
        formulation = TOTAL
        volumetric_locking_correction = false
        generate_output = 'cauchy_stress_yy cauchy_stress_xy strain_yy strain_xy' 
      [../]
    [../]
  [../]
[]

[AuxKernels]
  [./tempfuncaux]
    type = FunctionAux
    variable = temp
    function = temperature_load
  [../]
  [./eigenstrain_yy]
    type = RankTwoAux
    rank_two_tensor = eigenstrain
    variable = eigenstrain_yy
    index_i = 1
    index_j = 1
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_xx]
    type = RankTwoAux
    rank_two_tensor = eigenstrain
    variable = eigenstrain_xx
    index_i = 0
    index_j = 0
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_zz]
    type = RankTwoAux
    rank_two_tensor = eigenstrain
    variable = eigenstrain_zz
    index_i = 2
    index_j = 2
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_yy]
    type = RankTwoAux
    rank_two_tensor = total_strain
    variable = total_strain_yy
    index_i = 1
    index_j = 1
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_xx]
    type = RankTwoAux
    rank_two_tensor = total_strain
    variable = total_strain_xx
    index_i = 0
    index_j = 0
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_zz]
    type = RankTwoAux
    rank_two_tensor = total_strain
    variable = total_strain_zz
    index_i = 2
    index_j = 2
    execute_on = 'initial timestep_end'
  [../]
[]


[BCs]
  [./fix_disp_y_cube1]
    type = DirichletBC
    variable = disp_y
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]
  [./fix_disp_z_cube1]
    type = DirichletBC
    variable = disp_z
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]
  [./fix_disp_y_cube2]
    type = DirichletBC
    variable = disp_y
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]
  [./fix_disp_z_cube2]
    type = DirichletBC
    variable = disp_z
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]
  [./fix_disp_x_cube1_left]
    type = DirichletBC
    variable = disp_x
    boundary = 'left'
    value = 0.0
  [../]
  [./fix_disp_x_cube2_right]
    type = DirichletBC
    variable = disp_x
    boundary = 'right'
    value = 0.0
  [../]
[]


[Materials]
  [./elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 2.1e5
    poissons_ratio = 0.3
  [../]
  #[./small_stress]
    #type = ComputeFiniteStrainElasticStress
  [compute_stress]
    type = ComputeLagrangianLinearElasticStress  
  [../]
  [./thermal_expansion_strain]
    type = ComputeThermalExpansionEigenstrain
    stress_free_temperature = 200
    thermal_expansion_coeff = 17e-6 # thermal exxpansion coeff for copper
    temperature = temp
    eigenstrain_name = eigenstrain
  [../]
[]

[Executioner]
  type = Transient
  solve_type = 'PJFNK'

  l_max_its = 50
  nl_max_its = 50
  nl_rel_tol = 1e-12
  nl_abs_tol = 1e-10
  l_tol = 1e-9

  start_time = -0.0125
  n_startup_steps = 1
  end_time = 0.775
  dt = 0.1125
  dtmin = 0.0001
[]

[Outputs]
  exodus = true
[]

[Postprocessors]
  [./eigenstrain_xx]
    type = ElementAverageValue
    variable = eigenstrain_xx
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_yy]
    type = ElementAverageValue
    variable = eigenstrain_yy
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_zz]
    type = ElementAverageValue
    variable = eigenstrain_zz
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_xx]
    type = ElementAverageValue
    variable = total_strain_xx
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_yy]
    type = ElementAverageValue
    variable = total_strain_yy
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_zz]
    type = ElementAverageValue
    variable = total_strain_zz
    execute_on = 'initial timestep_end'
  [../]
  [./temperature]
    type = AverageNodalVariableValue
    variable = temp
    execute_on = 'initial timestep_end'
  [../]
[]
```

**Comment by GiudGiud** [comment]
Hello

The stitching probably failed?
What did the log say about the stitching operation? It should report on the number of nodes it managed to stitch

Guillaume

**Reply by AryGP** [reply]
This was the error:
unused parameter 'Mesh/cube1/subdomain_id'
unused parameter 'Mesh/cube2/subdomain_id'

**Reply by GiudGiud** [reply]
you are using an old version of moose I added this a month or so ago
update moose (and run tests if desired) with
```
conda update --all
cd ~/projects/moose
git fetch origin
git rebase origin/master
cd test
make -j6
./run_tests -j8
```

**Reply by AryGP** [reply]
is it supposed to be subdomain_ids instead of subdomain_id now?

**Reply by GiudGiud** [reply]
no
it s still subdomain_ids. subdomain_name is a new option
https://mooseframework.inl.gov/source/meshgenerators/GeneratedMeshGenerator.html

**Reply by AryGP** [reply]
ok.
```
[Mesh]
  [cube1]
    type = GeneratedMeshGenerator
    dim = 3
    nx = 2
    ny = 2
    nz = 2
    subdomain_name = 'cube1'
  []
  [cube2]
    type = GeneratedMeshGenerator
    dim = 3
    xmin = 1.2
    xmax = 2.2
    nx = 2
    ny = 2
    nz = 2
    subdomain_ids = '1'
    subdomain_name = 'cube2'
  []
  [stitch]
    type = StitchedMeshGenerator
    prevent_boundary_ids_overlap = False
    inputs = 'cube1 cube2'
    stitch_boundaries_pairs = 'right left'
  []
  #[combine]
   #type = CombinerGenerator
    #inputs = 'cube1 cube2'
  #[]
[]

[GlobalParams]
  displacements = 'disp_x disp_y disp_z'
  large_kinematics = false 
  eigenstrain_names = "thermal_contribution" 
[]

[AuxVariables]
  [./temp]
    initial_condition = 300.0
  [../]
  [./eigenstrain_yy]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./eigenstrain_xx]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./eigenstrain_zz]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./total_strain_yy]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./total_strain_xx]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./total_strain_zz]
    order = CONSTANT
    family = MONOMIAL
  [../]
[]

[Functions]
  [./temperature_load]
    type = ParsedFunction
    expression = t*(5000.0)+300.0
  [../]
[]

[Physics]
  [SolidMechanics]
    [QuasiStatic]
      [./all]
        strain = SMALL
        incremental = true
        add_variables = true
        eigenstrain_names = eigenstrain
        new_system = true
        formulation = TOTAL
        volumetric_locking_correction = false
        generate_output = 'cauchy_stress_yy cauchy_stress_xy strain_yy strain_xy' 
      [../]
    [../]
  [../]
[]

[AuxKernels]
  [./tempfuncaux]
    type = FunctionAux
    variable = temp
    function = temperature_load
  [../]
  [./eigenstrain_yy]
    type = RankTwoAux
    rank_two_tensor = eigenstrain
    variable = eigenstrain_yy
    index_i = 1
    index_j = 1
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_xx]
    type = RankTwoAux
    rank_two_tensor = eigenstrain
    variable = eigenstrain_xx
    index_i = 0
    index_j = 0
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_zz]
    type = RankTwoAux
    rank_two_tensor = eigenstrain
    variable = eigenstrain_zz
    index_i = 2
    index_j = 2
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_yy]
    type = RankTwoAux
    rank_two_tensor = total_strain
    variable = total_strain_yy
    index_i = 1
    index_j = 1
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_xx]
    type = RankTwoAux
    rank_two_tensor = total_strain
    variable = total_strain_xx
    index_i = 0
    index_j = 0
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_zz]
    type = RankTwoAux
    rank_two_tensor = total_strain
    variable = total_strain_zz
    index_i = 2
    index_j = 2
    execute_on = 'initial timestep_end'
  [../]
[]

[BCs]
  [./fix_disp_y_cube1]
    type = DirichletBC
    variable = disp_y
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]
  [./fix_disp_z_cube1]
    type = DirichletBC
    variable = disp_z
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]
  [./fix_disp_y_cube2]
    type = DirichletBC
    variable = disp_y
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]
  [./fix_disp_z_cube2]
    type = DirichletBC
    variable = disp_z
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]  
  [./fix_disp_x_cube1_left]
    type = DirichletBC
    variable = disp_x
    boundary = left
    value = 0.0
  [../]
  [./fix_disp_x_cube2_right]
    type = DirichletBC
    variable = disp_x
    boundary = right
    value = 0.0
  [../]  
[]

[Materials]
  [./elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    #youngs modulus and poissons ratio for COPPER
    youngs_modulus = 110e9 
    poissons_ratio = 0.34
  [../]
  #[./small_stress]
    #type = ComputeFiniteStrainElasticStress
  [compute_stress]
    type = ComputeLagrangianLinearElasticStress 
  [../]
  [./thermal_expansion_strain]
    type = ComputeThermalExpansionEigenstrain
    stress_free_temperature = 200
    #thermal exxpansion coeff for copper
    thermal_expansion_coeff = 17e-6 
    temperature = temp
    eigenstrain_name = eigenstrain
  [../]
[]

[Executioner]
  type = Transient
  solve_type = 'PJFNK'

  l_max_its = 50
  nl_max_its = 50
  nl_rel_tol = 1e-12
  nl_abs_tol = 1e-10
  l_tol = 1e-9

  start_time = -0.0125
  n_startup_steps = 1
  end_time = 0.775
  dt = 0.1125
  dtmin = 0.0001
[]

[Outputs]
  exodus = true
[]

[Postprocessors]
  [./eigenstrain_xx]
    type = ElementAverageValue
    variable = eigenstrain_xx
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_yy]
    type = ElementAverageValue
    variable = eigenstrain_yy
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_zz]
    type = ElementAverageValue
    variable = eigenstrain_zz
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_xx]
    type = ElementAverageValue
    variable = total_strain_xx
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_yy]
    type = ElementAverageValue
    variable = total_strain_yy
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_zz]
    type = ElementAverageValue
    variable = total_strain_zz
    execute_on = 'initial timestep_end'
  [../]
  [./temperature]
    type = AverageNodalVariableValue
    variable = temp
    execute_on = 'initial timestep_end'
  [../]
[]
```

The test starts glitching and says Solve did not Converge when I run this

**Reply by AryGP** [reply]
The test doesn't stop running

**Reply by GiudGiud** [reply]
> The test doesn't stop running

What does the simulation log look like?

**Reply by AryGP** [reply]
[testfail.mov.zip](https://github.com/user-attachments/files/16290161/testfail.mov.zip)

**Reply by AryGP** [reply]
This is what happens in the terminal

**Reply by GiudGiud** [reply]
text files can be uploaded directly, no need for a zip file

**Reply by AryGP** [reply]
I can't copy it, it's glitching . The run doesn't end

**Reply by AryGP** [reply]
I took a screen recording in the zip file to show you

**Reply by AryGP** [reply]
My tests run fine when I don't specify cube1 and cube2 in the boundary conditions

**Reply by GiudGiud** [reply]
The boundaries have the same names in the two cubes, and they have the same names
so you are blocking displacement everywhere all around both blocks, and your simulation is not converging.

use a RenameBoundaryGenerator on one cube to change the names of the boundaries on one of the cubes

**Reply by AryGP** [reply]
Ok, thanks. This run says I have cyclical dependencies but I'm not sure what else the input is supposed to be based on my boundary conditions. 
```
[Mesh]
  [cube1]
    type = GeneratedMeshGenerator
    dim = 3
    nx = 2
    ny = 2
    nz = 2
    subdomain_name = 'cube1'
  []
  [cube2]
    type = RenameBoundaryGenerator
    dim = 3
    xmin = 1.2
    xmax = 2.2
    nx = 2
    ny = 2
    nz = 2
    subdomain_ids = '1'
    subdomain_name = 'cube2'
    input = 'cube2'
    old_boundary = './fix_disp_x_cube2'
    new_boundary = './fix_disp_x_cube2_right ./fix_disp_x_cube2_left'
  []
  [stitch]
    type = StitchedMeshGenerator
    prevent_boundary_ids_overlap = False
    inputs = 'cube1 cube2'
    stitch_boundaries_pairs = 'right left'
  []
[]
```

**Reply by AryGP** [reply]
If I put something other than cube2 in the input then it says the requested MeshGenerator with name '...' was not found.

**Reply by AryGP** [reply]
I have the Rename Generator working but it doesn't use any of the dimension and xmin/max, and ny, nz, nx parameters. Is there a way I can still offset the cubes and not have them overlap?

[Mesh]
  [cube1]
    type = GeneratedMeshGenerator
    dim = 3
    nx = 2
    ny = 2 
    nz = 2
    subdomain_name = 'cube1'
  []
  [cube2]
    type = RenameBoundaryGenerator
     #dim = 3
     #xmin = 1.2
     #xmax = 2.2
     #nx = 2
     #ny = 2
     #nz = 2
     #subdomain_ids = '1'
     #subdomain_name = 'cube2'
    input = 'cube1'
    old_boundary = '0 1 2 3 4 5'
    new_boundary = 'left right top bottom front back'
  []
  [stitch]
    type = StitchedMeshGenerator
    prevent_boundary_ids_overlap = False
    inputs = 'cube1 cube2'
    stitch_boundaries_pairs = 'right left'
  []
[]

**Reply by AryGP** [reply]
Even when I define the xmin/xmax for cube 1, the cubes are still overlapping.

**Reply by GiudGiud** [reply]
> Ok, thanks. This run says I have cyclical dependencies but I'm not sure what else the input is supposed to be based on my boundary conditions.
> 
> ```
> [Mesh]
>   [cube1]
>     type = GeneratedMeshGenerator
>     dim = 3
>     nx = 2
>     ny = 2
>     nz = 2
>     subdomain_name = 'cube1'
>   []
>   [cube2]
>     type = RenameBoundaryGenerator
>     dim = 3
>     xmin = 1.2
>     xmax = 2.2
>     nx = 2
>     ny = 2
>     nz = 2
>     subdomain_ids = '1'
>     subdomain_name = 'cube2'
>     input = 'cube2'
>     old_boundary = './fix_disp_x_cube2'
>     new_boundary = './fix_disp_x_cube2_right ./fix_disp_x_cube2_left'
>   []
>   [stitch]
>     type = StitchedMeshGenerator
>     prevent_boundary_ids_overlap = False
>     inputs = 'cube1 cube2'
>     stitch_boundaries_pairs = 'right left'
>   []
> []
> ```

this is returning circular dependencies?

**Reply by GiudGiud** [reply]
>  Is there a way I can still offset the cubes and not have them overlap?

certainly.
Either use xmin/max  OR use a TransformGenerator in translate mode

**Reply by AryGP** [reply]
If I define input as cube2 then I get cyclic dependenices. 
If i define input as cube1 then both of the cubes overlap even with a redefined xmin/max. 

[Mesh]
  [cube1]
    type = GeneratedMeshGenerator
    dim = 3
    xmin = 1.5
    xmax = 2.5
    ymin = -0.5
    ymax = 0.5
    zmin = -0.5
    zmax = 0.5    
    nx = 2
    ny = 2
    nz = 2
    subdomain_name = 'cube1'
  []
  [cube2]
    type = RenameBoundaryGenerator
    input = 'cube1'
    old_boundary = '0 1 2 3 4 5'
    new_boundary = 'left right top bottom front back'
  []
  [stitch]
    type = StitchedMeshGenerator
    prevent_boundary_ids_overlap = False
    inputs = 'cube1 cube2'
    stitch_boundaries_pairs = 'right left'
  []
[]

[GlobalParams]
  displacements = 'disp_x disp_y disp_z'
  large_kinematics = false 
  eigenstrain_names = "thermal_contribution" 
[]

[AuxVariables]
  [./temp]
    initial_condition = 300.0
  [../]
  [./eigenstrain_yy]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./eigenstrain_xx]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./eigenstrain_zz]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./total_strain_yy]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./total_strain_xx]
    order = CONSTANT
    family = MONOMIAL
  [../]
  [./total_strain_zz]
    order = CONSTANT
    family = MONOMIAL
  [../]
[]

[Functions]
  [./temperature_load]
    type = ParsedFunction
    expression = t*(5000.0)+300.0
  [../]
[]

[Physics]
  [SolidMechanics]
    [QuasiStatic]
      [./all]
        strain = SMALL
        incremental = true
        add_variables = true
        eigenstrain_names = eigenstrain
        new_system = true
        formulation = TOTAL
        volumetric_locking_correction = false
        generate_output = 'cauchy_stress_yy cauchy_stress_xy strain_yy strain_xy' 
      [../]
    [../]
  [../]
[]

[AuxKernels]
  [./tempfuncaux]
    type = FunctionAux
    variable = temp
    function = temperature_load
  [../]
  [./eigenstrain_yy]
    type = RankTwoAux
    rank_two_tensor = eigenstrain
    variable = eigenstrain_yy
    index_i = 1
    index_j = 1
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_xx]
    type = RankTwoAux
    rank_two_tensor = eigenstrain
    variable = eigenstrain_xx
    index_i = 0
    index_j = 0
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_zz]
    type = RankTwoAux
    rank_two_tensor = eigenstrain
    variable = eigenstrain_zz
    index_i = 2
    index_j = 2
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_yy]
    type = RankTwoAux
    rank_two_tensor = total_strain
    variable = total_strain_yy
    index_i = 1
    index_j = 1
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_xx]
    type = RankTwoAux
    rank_two_tensor = total_strain
    variable = total_strain_xx
    index_i = 0
    index_j = 0
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_zz]
    type = RankTwoAux
    rank_two_tensor = total_strain
    variable = total_strain_zz
    index_i = 2
    index_j = 2
    execute_on = 'initial timestep_end'
  [../]
[]

[BCs]
  [./fix_disp_y_cube1]
    type = DirichletBC
    variable = disp_y
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]
  [./fix_disp_z_cube1]
    type = DirichletBC
    variable = disp_z
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]
  [./fix_disp_y_cube2]
    type = DirichletBC
    variable = disp_y
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]
  [./fix_disp_z_cube2]
    type = DirichletBC
    variable = disp_z
    boundary = 'left right top bottom front back'
    value = 0.0
  [../]  
  [./fix_disp_x_cube1_left]
    type = DirichletBC
    variable = disp_x
    boundary = left
    value = 0.0
  [../]
  [./fix_disp_x_cube2_right]
    type = DirichletBC
    variable = disp_x
    boundary = right
    value = 0.0
  [../]  
[]

[Materials]
  [./elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    #youngs modulus and poissons ratio for COPPER
    youngs_modulus = 110e9 
    poissons_ratio = 0.34
    #block = 'cube1 cube2'    
  [../]
  #[./small_stress]
    #type = ComputeFiniteStrainElasticStress
  [compute_stress]
    type = ComputeLagrangianLinearElasticStress 
    #block = 'cube1 cube2'
  [../]
  [./thermal_expansion_strain]
    type = ComputeThermalExpansionEigenstrain
    stress_free_temperature = 200
    #thermal exxpansion coeff for copper
    thermal_expansion_coeff = 17e-6 
    temperature = temp
    eigenstrain_name = eigenstrain
    #block = 'cube1 cube2'
  [../]
[]

[Executioner]
  type = Transient
  solve_type = 'PJFNK'

  l_max_its = 50
  nl_max_its = 50
  nl_rel_tol = 1e-12
  nl_abs_tol = 1e-10
  l_tol = 1e-9

  start_time = -0.0125
  n_startup_steps = 1
  end_time = 0.775
  dt = 0.1125
  dtmin = 0.0001
[]

[Outputs]
  exodus = true
[]

[Postprocessors]
  [./eigenstrain_xx]
    type = ElementAverageValue
    variable = eigenstrain_xx
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_yy]
    type = ElementAverageValue
    variable = eigenstrain_yy
    execute_on = 'initial timestep_end'
  [../]
  [./eigenstrain_zz]
    type = ElementAverageValue
    variable = eigenstrain_zz
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_xx]
    type = ElementAverageValue
    variable = total_strain_xx
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_yy]
    type = ElementAverageValue
    variable = total_strain_yy
    execute_on = 'initial timestep_end'
  [../]
  [./total_strain_zz]
    type = ElementAverageValue
    variable = total_strain_zz
    execute_on = 'initial timestep_end'
  [../]
  [./temperature]
    type = AverageNodalVariableValue
    variable = temp
    execute_on = 'initial timestep_end'
  [../]
[]

<img width="624" alt="image" src="https://github.com/user-attachments/assets/83df226b-37d8-4420-a444-24ed683c9f9e">

**Reply by GiudGiud** [reply]
this is behaving as expect
```
[cube2]
type = RenameBoundaryGenerator
input = 'cube1'
old_boundary = '0 1 2 3 4 5'
new_boundary = 'left right top bottom front back'
[]
```
this means cube 2 is exactly the same as cube 1 except for the names of the boundaries.
you need to use more than one mesh generator to define cube2 to be different from cube1.
Maybe a GeneratedMeshGenerator then a RenameBoundariesGenerator if you want both :
- different location
- different names of sidesets

**Reply by AryGP** [reply]
I see, so I still need to have my original 2 mesh cubes then rename the boundary of just the second cube. Something like this?
[Mesh]
  [cube1]
    type = GeneratedMeshGenerator
    dim = 3
    xmin = 1.5
    xmax = 2.5
    ymin = -0.5
    ymax = 0.5
    zmin = -0.5
    zmax = 0.5    
    nx = 2
    ny = 2
    nz = 2
    subdomain_name = 'cube1'
  []
  [cube2]
    type = RenameBoundaryGenerator
    dim = 3
    xmin = 1.2
    xmax = 2.2
    nx = 2
    ny = 2
    nz = 2
    subdomain_ids = '1'
    subdomain_name = 'cube2'
 [] 
   [cube_boundary]
    input = 'cube2'
    old_boundary = '0 1 2 3 4 5'
    new_boundary = 'left right top bottom front back'
  []
  [stitch]
    type = StitchedMeshGenerator
    prevent_boundary_ids_overlap = False
    inputs = 'cube1 cube2'
    stitch_boundaries_pairs = 'right left'
  []
[]

**Reply by AryGP** [reply]
Not sure if my old_boundary and new_boundary are defined correctly, though.

**Reply by GiudGiud** [reply]
this is wrong
```
[cube2]
type = RenameBoundaryGenerator
dim = 3
xmin = 1.2
xmax = 2.2
nx = 2
ny = 2
nz = 2
subdomain_ids = '1'
subdomain_name = 'cube2'
[]
[cube_boundary]
input = 'cube2'
old_boundary = '0 1 2 3 4 5'
new_boundary = 'left right top bottom front back'
[]
```

it should be something like
```
[cube2]
type = GeneratedMeshGenerator
dim = 3
xmin = 1.2
xmax = 2.2
nx = 2
ny = 2
nz = 2
subdomain_ids = '1'
subdomain_name = 'cube2'
[]
[cube_boundary]
type = RenameBoundaryGenerator
input = 'cube2'
old_boundary = '0 1 2 3 4 5'
new_boundary = 'left right top bottom front back'
[]
```
